src/extension.ts

```diff
@@ -64,6 +64,13 @@ export async function activate(context: ExtensionContext): Promise<XMLExtensionA
 
   languageClient = await startLanguageClient(context, serverOptions, logfile, externalXmlSettings, requirementsData);
 
+  const onDidGrantWorkspaceTrust = (workspace as any).onDidGrantWorkspaceTrust;
+  if (onDidGrantWorkspaceTrust !== undefined) {
+    context.subscriptions.push(onDidGrantWorkspaceTrust(() => {
+      commands.executeCommand("xml.downloadExternalResources.enabled", true);
```

Comment:
I think the proper code should be the same thanblock code at https://github.com/redhat-developer/vscode-xml/blob/ed8e6bb8c4bc686144870d58f7dfb7f9dd4715e5/src/client/xmlClient.ts#L150

I suggest that you move this blick in a private function inXMLClient.ts because I noticed that this code is used in:

 * https://github.com/redhat-developer/vscode-xml/blob/master/src/client/xmlClient.ts#L150
 * https://github.com/redhat-developer/vscode-xml/blob/ed8e6bb8c4bc686144870d58f7dfb7f9dd4715e5/src/client/xmlClient.ts#L101

and now we your onDidGrantWorkspaceTrust too.

Reply:
`onDidGrantWorkspaceTrust` as its name implies is only triggered when trust is granted (so not when something changes to untrusted). `(workspace as any).isTrusted` is `true` in the callback though.

Reply:
OK so it means my suggestion was wrong and we need to update settings as Alex did the first time?

Reply:
Was just going to say. the setting, `xml.downloadExternalResources.enabled` as `true` needs to be injected. The problem with the previous path was that it isn't a command to be executed.

Reply:
Can we inject the setting change here:
https://github.com/redhat-developer/vscode-xml/blob/ed8e6bb8c4bc686144870d58f7dfb7f9dd4715e5/src/settings/settings.ts#L227-L232
since we generate new XML settings from the `onDidGrantWorkspaceTrust` event?

Reply:
If we set it to true based on the the workspace trust state, then we can never have a case where a user sets it to false, which might be a disadvantage.

Let's just manually set it in the callback.